internal/muxado/buffer.go

```diff
@@ -68,7 +98,24 @@ func (b *inboundBuffer) Read(p []byte) (n int, err error) {
 			err = b.err
 			break
 		}
+
+		// If the deadline is set, we need to take it into account
+		if !b.deadline.IsZero() {
```

Comment:
That's true, but since `Read` is pretty much always called in a loop, it'll just get called again immediately after and return the error. 

```go
n, err = conn.Read(buf) // non-zero, ErrDeadlineExceeded

// vs

n, err = conn.Read(buf) // non-zero, nil
n, err = conn.Read(buf) // 0, ErrDeadlineExceeded
```

Feels like they're more or less the same thing, but with less opportunity for bad consumers to forget to handle the non-zero/non-nil-error case.

I guess there's the possibility of the buffer getting filled faster than the reader can drain it, which would cause the deadline to never get seen?

Reply:
OK, moved the deadline check to the start of the loop